src/main/java/com/projectkorra/core/util/ParticleUtil.java

```diff
@@ -0,0 +1,603 @@
+	}
+	
+	/**
+	 * Spawns an amount of particles that move from the given Location in the given direction at the given speed 
+	 * @param particle a particle which can have direction applied to it
+	 * @param direction which direction to move the particles
+	 * @param speed how fast to move the particle
+	 * @param loc where the particles spawn at
+	 * @param amount how many particles to spawn
+	 */
+	public static void spawnWithDirection(DirectionalParticle particle, Vector direction, double speed, Location loc, int amount) {
+		spawnWithDirection(particle, direction, speed, loc, amount, 0);
+	}
+	
+	/**
+	 * Spawns a single particle that moves from the given Location in the given direction at the given speed 
+	 * @param particle a particle which can have direction applied to it
+	 * @param direction which direction to move the particles
+	 * @param speed how fast to move the particle
+	 * @param loc where the particles spawn at
+	 */
+	public static void spawnWithDirection(DirectionalParticle particle, Vector direction, double speed, Location loc) {
+		spawnWithDirection(particle, direction, speed, loc, 1);
+	}
+	
+	/**
+	 * Spawns a single particle that moves from the given Location in the given direction using the magnitude of the direction vector as the speed
+	 * @param particle a particle which can have direction applied to it
+	 * @param direction which direction to move the particles
+	 * @param loc where the particles spawn at
+	 */
+	public static void spawnWithDirection(DirectionalParticle particle, Vector direction, Location loc) {
+		spawnWithDirection(particle, direction, direction.length(), loc);
+	}
+	
+	public static enum NoteColor {
+		GREEN, 
+		LIME, 
+		YELLOW, 
+		GOLD, 
+		ORANGE, 
+		RED_ORANGE,
+		RED,
+		CRIMSON,
+		FUSCHIA,
+		MAGENTA,
+		PINK,
+		PURPLE,
+		VIOLET,
+		INDIGO,
+		NAVY,
+		BLUE,
+		LIGHT_BLUE,
+		SKY_BLUE,
+		AQUA,
+		TEAL,
+		MALACHITE,
+		VIVID_GREEN,
+		DULL_GREEN,
+		NEON_GREEN,
+		GREEN_YELLOW;
+	}
+	
+	public static enum DirectionalParticle {
+		EXPLOSION_NORMAL (Particle.EXPLOSION_NORMAL),
+		FIREWORKS_SPARK (Particle.FIREWORKS_SPARK),
+		WATER_BUBBLE (Particle.WATER_BUBBLE),
+		WATER_WAKE (Particle.WATER_WAKE),
+		CRIT (Particle.CRIT),
+		CRIT_MAGIC (Particle.CRIT_MAGIC),
+		SMOKE_NORMAL (Particle.SMOKE_NORMAL),
+		SMOKE_LARGE (Particle.SMOKE_LARGE),
+		PORTAL (Particle.PORTAL),
+		ENCHANTMENT_TABLE (Particle.ENCHANTMENT_TABLE),
+		FLAME (Particle.FLAME),
+		CLOUD (Particle.CLOUD),
+		DRAGON_BREATH (Particle.DRAGON_BREATH),
+		END_ROD (Particle.END_ROD),
+		DAMAGE_INDICATOR (Particle.DAMAGE_INDICATOR),
+		TOTEM (Particle.TOTEM),
+		SPIT (Particle.SPIT),
+		SQUID_INK (Particle.SQUID_INK),
+		BUBBLE_POP (Particle.BUBBLE_POP),
+		BUBBLE_COLUMN_UP (Particle.BUBBLE_COLUMN_UP),
+		NAUTILUS (Particle.NAUTILUS),
+		CAMPFIRE_COZY_SMOKE (Particle.CAMPFIRE_COSY_SMOKE), CAMPFIRE_COSY_SMOKE (Particle.CAMPFIRE_COSY_SMOKE),
```

Comment:
Did you intend for these to be on the same line?

Reply:
yes since they're the same particle, but i figured having both spellings of cosy / cozy would be good